osu.Game.Modes.Taiko/Objects/BarLine.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) 2007-2017 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using osu.Game.Beatmaps;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace osu.Game.Modes.Taiko.Objects
+{
+    public class BarLine
+    {
+        public double StartTime;
+        public double PreEmpt;
+
+        public void SetDefaultsFromBeatmap(Beatmap beatmap)
+        {
+            // Don't ask... Old osu! had a random multiplier here, that we now have to multiply everywhere
+            float fudgeFactor = 1.4f;

```

Comment:
With that being said I have been talking to ppy and we decided that hitobjects should never be passed the beatmap as they are in here. So this method will probably be gone sometime really soon.